sdk/compiler/damlc/daml-stdlib-src/DA/Internal/Template/Functions.daml

```diff
@@ -110,10 +110,6 @@ type Choice t c r =
   , HasExercise t c r
   , HasToAnyChoice t c r
   , HasFromAnyChoice t c r
-#ifdef DAML_CHOICE_FUNCS
-  , HasChoiceController t c
-  , HasChoiceObserver t c
-#endif -- DAML_CHOICE_FUNCS
```

Comment:
Hey @moisesackerman-da why do we need those in the `Choice` alias.
With this I cannot plug LF 1.dev with 1.16.

Reply:
they're not strictly needed, I think the idea was just that users could use them with a `Choice t c r` constraint instead of the longer `HasChoiceController t c`/`HasChoiceObserver t c`.

> With this I cannot plug LF 1.dev with 1.16

I think that means that some instances of LF 1.dev have to stay as 1.dev instead of being replaced with 1.16, since they need the `DAML_CHOICE_FUNCS` feature. What breaks if you keep these lines?

Reply:
for instance the daml script
```
internalTypedExerciseCmd : forall t c r. Choice t c r => ContractId t -> c -> Bool -> Commands r
``` 
compiled in 1.dev will require instances for  `HasChoiceController t c`  `HasChoiceObserver t c`, but they are not generated for <= 1.16.

Reply:
I don't totally follow whats happening here, if the constraints in daml3-script need changing though, thats fine?

Reply:
Ah, I see it now. If package `A` is compiled with LF 1.16, and I import it as a data-dependency from a package `B` compiled with LF 1.dev, I won't be able to call e.g. `exerciseCmd : forall t c r. Choice t c r => ContractId t -> c -> Commands r` on a template from `A`, because `Choice t c r` now includes a constraint corresponding to an instance that was never generated for `A` templates, and data-dependencies does not add that instance (and I don't think it could). 

tl;dr: let's remove these from `type Choice`, thank you Rémy